platform/linux-generic/odp_crypto_openssl.c

```diff
@@ -1568,26 +1446,14 @@ odp_crypto_alg_err_t aes_ccm_encrypt(odp_packet_t pkt,
 	EVP_CIPHER_CTX *ctx = local.cipher_ctx[session->idx];
 	const uint8_t *aad_head = param->aad_ptr;
 	uint32_t aad_len = session->p.auth_aad_len;
-	void *iv_ptr;
+	void *iv_ptr = param->cipher_iv_ptr;
 	int dummy_len = 0;
 	int cipher_len;
 	uint32_t in_len = param->cipher_range.length;
 	uint8_t data[in_len];
 	uint8_t block[EVP_MAX_MD_SIZE];
 	int ret;
 
-#if ODP_DEPRECATED_API
-	if (param->cipher_iv_ptr)
-		iv_ptr = param->cipher_iv_ptr;
-	else if (session->p.cipher_iv.data)
-		iv_ptr = session->cipher.iv_data;
-	else
-		return ODP_CRYPTO_ALG_ERR_IV_INVALID;
-#else
-	iv_ptr = param->cipher_iv_ptr;
-	_ODP_ASSERT(session->p.cipher_iv_len == 0 || iv_ptr != NULL);
-#endif
-
 	EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_CCM_SET_TAG,
 			    session->p.auth_digest_len, NULL);
 	EVP_EncryptInit_ex(ctx, NULL, NULL, session->cipher.key_data, iv_ptr);
```

Comment:
`param->cipher_iv_ptr` could be used here directly (instead of via `iv_ptr`)?

(Same in `aes_ccm_decrypt`.)

Reply:
Done in v2.